src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int NUMBER_OF_FIGURES_NAME = 5;
+    private static final Random random = new Random();
+    private static final String DEFAULT_COLOR = "white";
+    private static final String DEFAULT_NAME = "circle";
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MAX_AREA_LIMIT = 100;
+
+    private String getRandomFigureName() {
+        return FigureName.values()[random.nextInt(NUMBER_OF_FIGURES_NAME)].name();
+    }
+
+    public Figure getRandomFigure() {
+        String randomColor = getRandomColor();
+        // String randomFigureName = getRandomFigureName();
+        double randomArea = random.nextDouble(MAX_AREA_LIMIT);

```

Comment:
create separate method to generate really random area and use it for creating each type of figure